components/textbox/index.js

```diff
@@ -4,22 +4,46 @@
 import {bindActionCreators} from 'redux';
 import {connect} from 'react-redux';
 
+import {getCurrentChannel} from 'mattermost-redux/selectors/entities/channels';
+
+import {getGroupsForReference, getGroupsAssociatedToChannelForReference, getGroupsAssociatedToTeamForReference} from 'mattermost-redux/selectors/entities/groups';
+
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+
 import {getCurrentUserId, makeGetProfilesInChannel, makeGetProfilesNotInChannel} from 'mattermost-redux/selectors/entities/users';
 
 import {autocompleteUsersInChannel} from 'actions/views/channel';
 import {autocompleteChannels} from 'actions/channel_actions';
 
 import Textbox from './textbox.jsx';
 
-const makeMapStateToProps = () => {
+const makeMapStateToProps = (state, ownProps) => {
     const getProfilesInChannel = makeGetProfilesInChannel();
     const getProfilesNotInChannel = makeGetProfilesNotInChannel();
 
-    return (state, ownProps) => ({
+    const team = getCurrentTeam(state);
+    const channel = getCurrentChannel(state);
+
+    let autocompleteGroups;
+
+    if(team && team.group_constrained && channel && channel.group_constrained) {
```

Comment:
I think you can do this because we have [optional chaining enabled in our babel config](https://github.com/mattermost/mattermost-webapp/blob/master/babel.config.js#L33).
```suggestion
    if(team?.group_constrained && channel?.group_constrained) {
```

Reply:
moved this logic to redux, where for some reason I cannot do `?.`, so it'll have to stay this way fo now.